spec/hosts/nodes/foreman.cp.lsst.org_spec.rb

```diff
@@ -0,0 +1,162 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe 'foreman.cp.lsst.org', :site do
+  on_supported_os.each do |os, facts|
+    context "on #{os}" do
+      let(:facts) do
+        facts.merge(
+          fqdn: 'foreman.cp.lsst.org',
+        )
+      end
+
+      let(:node_params) do
+        {
+          role: 'foreman',
+          site: 'cp',
+          cluster: 'corecp',
+        }
+      end
+
+      let(:manifest) do
```

Comment:
In puppet, the `manifest` is the source code that is "compiled" into a `catalog`.  The goal of the role unit test(s) is to inspect the `catalog` that is created from compiling the profiles together with the hiera data.  The block of code in the `:manfiest` string here is adding `dhcp::pool` defined types into the manifest for this spec example but it doesn't help determine if the hiera data is resulting in the desired objects in the catalog.

What I suggest you do instead is inspect the catalog for `dhcp::pool` defined types. E.g.

(untested example)
```ruby
  it do
    is_expected.to contain_dhcp__pool('it_gss').with(
       network: '139.229.160.0',
       ...
    )
  end
```

Reply:
Also note that `foreman.cp` role is already being tested here: https://github.com/lsst-it/lsst-itconf/blob/master/spec/hosts/roles/foreman_spec.rb#L255-L269  The distinction between a "role" test a "node" test is that "node" is covering things they are truly specific to a single node. IOW - the data from the `hierdata/node` layer.

Reply:
Sounds perfect! I'll apply the changes. Regarding the role test vs node test, do you want me to add and exclude foreman/cp from the role test?